test/lib/omniauth/strategies/openid_connect_test.rb

```diff
@@ -65,6 +65,30 @@ def test_logout_phase_with_discovery_and_post_logout_redirect_uri
         strategy.other_phase
       end
 
+      def test_logout_phase_with_discovery_and_custom_post_logout_redirect_uri
+        expected_redirect = 'https://example.com/logout?post_logout_redirect_uri=https%3A%2F%2Fmysite.com%2Fpost_logout'
+        strategy.options.client_options.host = 'example.com'
+        strategy.options.discovery = true
+
+        issuer = stub('OpenIDConnect::Discovery::Issuer')
+        issuer.stubs(:issuer).returns('https://example.com/')
+        ::OpenIDConnect::Discovery::Provider.stubs(:discover!).returns(issuer)
+
+        config = stub('OpenIDConnect::Discovery::Provder::Config')
+        config.stubs(:authorization_endpoint).returns('https://example.com/authorization')
+        config.stubs(:token_endpoint).returns('https://example.com/token')
+        config.stubs(:userinfo_endpoint).returns('https://example.com/userinfo')
+        config.stubs(:jwks_uri).returns('https://example.com/jwks')
+        config.stubs(:end_session_endpoint).returns('https://example.com/logout')
+        ::OpenIDConnect::Discovery::Provider::Config.stubs(:discover!).with('https://example.com/').returns(config)
+
+        request.stubs(:path_info).returns('/auth/openidconnect/logout')
+        request.stubs(:params).returns('post_logout_redirect_uri' => 'https://mysite.com/post_logout')
+
+        strategy.expects(:redirect).with(expected_redirect)
```

Comment:
Shouldn't we test that the option is `nil` by default, when no param is provided?